### PR TITLE
Leave apply

### PR DESCRIPTION
**Pull request by excellencetechnologies**
Pull Request

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -48,79 +49,7 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
...
-                            }
-                        })
-
-                    }
-                }
-            }
+        request_.request(message, function (req, response, msg) {
         });
     } else if (message.text == 'apply' || date == true || from != '') {
         if (message.text == 'apply') {
```
Comment (reactexcel) [inline routes/index.js]: all code after apply should go in a seperate file named

services/leave/apply.js

File: routes/index.js [hunk 2]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
             rtm.sendMessage('reason', dm.id);
         } else if (from != '' && to != '' && reason == '') {
             reason = message.text;
-            rtm.sendMessage('your leave application has been submitted', dm.id);
+            var id = message.user;
+            var myFrom = from.substring(8, 10);
```
Comment (reactexcel) [inline routes/index.js]: use momentjs library to find out day

File: routes/index.js [hunk 3]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
...
+            var myTo = to.substring(8, 10);
+            var num = myTo * 1 - myFrom * 1;
+            var number_of_day = num + 1;
```
Comment (reactexcel) [inline routes/index.js]: find moment.js to find number of days between dates

File: service/leaveApply.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+var request = require('request');
+
+exports.leaveApply = function (id, from, to, number_of_day, reason, callback) {
+    request({
+        url: 'http://excellencemagentoblog.com/slack_dev/hr/attendance/API_HR/api.php', //URL to hit
```
Comment (reactexcel) [inline service/leaveApply.js]: this url should to config file

File: routes/index.js [hunk 5]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
...
+            leave_.leaveApply(id, from, to, number_of_day, reason, function (response) {
+                if (response == 0) {
+                    rtm.sendMessage('your leave application has been submitted', dm.id);
```
Comment (reactexcel) [inline routes/index.js]: Your leave has been submitted approval!

File: routes/index.js [hunk 6]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
...
+                } else {
+                    rtm.sendMessage('your leave application has not been submitted', dm.id);
```
Comment (reactexcel) [inline routes/index.js]: Oops! Some problem occurred. We are looking into it. In the mean time you can use HR system to apply your leave

File: service/request.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+var request = require('request');
+const request_send = require('../service/request_send');
+
+exports.request = function (message) {
+    var Approved = [];
+    var Pending = [];
+    var cancelled = [];
+    request({
```
Comment (reactexcel) [inline service/request.js]: rtm.sendMessage 

please wait...

File: routes/index.js [hunk 8]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
             rtm.sendMessage('reason', dm.id);
```
Comment (reactexcel) [inline routes/index.js]: ask from first and then to

File: routes/index.js (hunk 3 above)
Comment (reactexcel) [inline routes/index.js]: put validation that days are positive number

File: routes/index.js [hunk 9]
```diff
@@ -134,14 +63,25 @@ rtm.on(RTM_EVENTS.MESSAGE, function (message) {
...
         } else if (from != '' && to != '' && reason == '') {
             reason = message.text;
-            rtm.sendMessage('your leave application has been submitted', dm.id);
+            var id = message.user;
```
Comment (reactexcel) [inline routes/index.js]: if user puts invalid date ask again for date.

Oops! Unable to understand. Please provide date in proper format (MM-DD-YYYY)

File: service/leaveApply.js (hunk 4 above)
Comment (excellencetechnologies) [inline service/leaveApply.js]: Done

File: service/request.js (hunk 7 above)
Comment (excellencetechnologies) [inline service/request.js]: Done

File: routes/index.js (hunk 6 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

File: routes/index.js (hunk 5 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

File: routes/index.js (hunk 3 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

File: routes/index.js (hunk 2 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

File: routes/index.js (hunk 8 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

File: routes/index.js (hunk 9 above)
Comment (excellencetechnologies) [inline routes/index.js]: Done

**Pull request merged**